content/en/docs/Responses/ReplayGain.md

```diff
@@ -0,0 +1,45 @@
+---
```

Comment:
For replay gain it may be worth making a note somewhere that if a track is missing replayGain info, the server must return `"replayGain": {}` as the default and **not** `replayGain: { "trackGain": 0, "albumGain": 0, ... }`

Reply:
This does not change much on client side handling, but since the fields are optional they by default should be omitted since it's a new field.

Reply:
It does change things on the client side. A trackGain or albumGain of 0 tells the client that no replayGain processing should be applied. Missing replayGain info means the client can use a default gain (like -8 dB, since most popular music has a replayGain value of around -8 to -10).

Reply:
Well on my client the users requested and wanted a base gain for all then replay gains value added to it. Is there clients that offers a base replay gain only when there's no replay gain in tags?
Anyway no problem to be more clear about it, just wondering the usage in case I need to implement that in Symfonium.

Reply:
It's called fallback gain in this case and I want to add it to Supersonic eventually, but I will have to change my replayGain implementation a little bit. The usefulness of it is that most popular music tracks have a large negative gain when they are scanned, and if you have an unscanned track it will end up VERY loud relative to the others, unless you have a default fallback gain in place like -8 dB

Reply:
Hum no one ever asked that the base replay gain was enough. Will see if someone ask for that one day.
But if it's common should we not have something exposed by the server? Since the server can scan all the files and help finding a proper fallback gain no?

Reply:
It would be something that should be configured in the client settings, IMO

Reply:
![image](https://github.com/opensubsonic/open-subsonic-api/assets/5303666/0ad3e016-b5ab-4b94-908b-ad3efb9641fd)

ReplayGain settings from Supersonic. Eventually I may add a field for Fallback Gain eg. -8 db

Reply:
Yes so you do not have it yet ;) I have the same, but as said the server can scan all the files, get the peaks and expose a proper value. It's absurd to have it configured on all client side if it's always necessary it should be configured once on the server. I don't think that it have any drawback to anticipate and add a fallbackGain to the response.

Reply:
I guess it's a difference in philosophy. I was thinking the fallback gain is a user preference, and since it's a playback detail should be configured on the client. But I suppose there's a case to be made for the server to take the average of all the replayGain values across the entire library and then expose that value as `fallbackGain` to the client, for files that are missing their own replayGain tags.

Reply:
It is an user preference, but userRating and starred are user preferences too but we store them on the server so that every clients have them up to date and it support multiple users.
A fallbackGain here would also be user related since the queries are tied to the user and the server can either scan or offer a setting for the user to configure (Or I guess as 99% of the server will just ignore the field :p)
I'm adding it, even if no one implement it, it's still an useful reminder to clients that users may want that.

Reply:
In that case I would think it would make sense to store all the user's ReplayGain preferences on the server, no? And a new API for `getUserReplayGainPreferences` that includes the preferred mode (track, album, or none), preamp, falback gain, and preventClipping

Reply:
that's what I disagree with, fallbackGain being an immutable value tied to the songs - **unless** it's not a user setting but computed by the server as an average based on the rest of the library. In which case I'm happy for it to be a server-supplied value. But I think we're just talking past each other now and can wait for others to chime in :)

Reply:
Well this spam a little the others, but this is interesting discussion. I do not grasp why that value is not tied to the songs even if supplied as a user setting?
The goal is to have those songs not sound too loud when there's no gain, so it's by definition tied to all the songs and replay gain on some of the songs.
That difference of value between tagged songs and not tagged songs is something immutable, whatever the device that plays the media the difference in DB will be the same.

On the other end preAmp and clipping are personal choices that depends on where the sounds goes so does not fit a server side settings.

Or is there something else I'm missing here? Even with fallbackGain you'd still apply the preAmp (That could be negative to give head room)

Reply:
In a way you're right, even as a user setting it's the user's "best guess" as to what the average trackGain in their library is. But just as a user, IMO it would be confusing to put all my replaygain settings in the client, except fallback gain which I have to go configure on the server.

But as you mentioned the server knows all the replaygain values, so the server's "best guess" is going to be better than the user's best guess, which is why there's a case that the server should return fallback gain as the average of all the tracks that do have trackGain. And the server could do something even smarter if it wanted to, like if there is a single album track missing replayGain, the fallbackgain for that track could be set to the average for the rest of the album's tracks (or the album gain value for the album). Or it could be done by genre, because classical tracks with a missing replayGain value should probably be compared to other classical tracks with replaygain, rather than rock/pop. Or, the server could even run a replayGain scan using ffmpeg on files missing it, and just store actual replaygain values in its database and have no need for fallbackGain at all :)

All this is to say that the server is better at supplying a reasonable fallbackGain than the user would be, so if it's going to be a server value, it should be computed by the server IMO.

Reply:
I would say that the client should still have a way to set fallback gain (for example, a library with no replaygain tags whatsoever), but defaulting to the server computed value is probably fine (and how exactly it's done can be left to the individual servers). The _ideal_ (at least personally) would be server options to set the gain mode and preamp, but perhaps that's better as an enhancement later.